core/database/foxx/api/user_router.js

```diff
@@ -516,9 +519,47 @@ router.get('/token/set', function(req, res) {
                         refresh: req.queryParams.refresh,
                         expiration: Math.floor(Date.now() / 1000) + req.queryParams.expires_in
                     };
-                    g_db._update(user_id, obj, {
-                        keepNull: false
-                    });
+                    if (token_type && other_token_data) {
+
+                        // NOTE: the edge is only defined for the `u` collection and the `globus_coll` collection.
+                        //  The getUserFromClientID method shows that the user account can also be stored in a UUID collection
+                        //  and in the future support domain accounts. For now, we must ensure that the edge is u to globus_coll
+                        const user_doc = g_db.u.exists(user_id);    // TODO: user doc only necessary for edge key, user_doc._key must not have invalid character '/'
+                                                                    //  which should be kicked out by arango
+                        if (!user_doc) {
+                            throw [g_lib.ERR_INVALID_PARAM, "Error writing to globus_token. No such user in u collection '" + user_id + "'"];
+                        }
+
+                        // find or insert collection
+                        const collection_search_key = other_token_data;
+                        let globus_collection = g_db.globus_coll.exists({_key: collection_search_key});
+                        if (!globus_collection) {
+                            // TODO: what other information should live on the globus_coll document?
```

Comment:
I want a discussion here if possible. I'm not sure what we should be doing with the `globus_coll` as it is not yet defined in any place I could find.
We can make moves to store important data now, or pass that on to later - in which case I should remove the additional field.

Reply:
Trying to get clarification here, does the `globus_coll` doc have:
```javascript
    class GlobusCollection {
        +string _key
        +string name
    }
```

Maybe...
```javascript
description: "Brief description of the collection.",
owner: "User or entity that owns the collection.",
created_at: "Timestamp of when the collection was created.",
updated_at: "Timestamp of the last update to the collection.",
```

Reply:
Currently, from what I can tell, `globus_coll` is totally unused. The work being done here is defining what the document's structure will look like, with a lot of room for change down the road.

I think you're right that these fields would be great adds, I am just unsure how I would get that additional information as of right now. Except for `created_at` and `updated_at` which I assume are metadata for the records themselves.

Reply:
@AronPerez  makes a good suggestion about update time and create time. The data_router also does this for records for consistency consider using the same language.

Below is from the data_router. I'm not really a fan of ct and ut, but its probably more important to be consistent.
```
    var time = Math.floor(Date.now() / 1000),
        obj = {
            size: 0,
            ct: time,
            ut: time,
            owner: owner_id,
            creator: client._id
        },
 ```
 
 In terms of other information, this is what I can think of. 
 
 ```
 _key: "This should be the uuid of the collection",
 _id: "globus_coll/<uuid of the collection",
 scopes: "",
 ca: Math.floor(Date.now() / 1000),
 ua: Math.floor(Date.now() / 1000), 
 type: "<mapped, guest>", - will need to know this because it effects consent flows. There may be other collection types, I'm not yet sure if it matters if it is a personal or server collection type, that also might be important.
 ha_enabled: "true/false"  - will also need to know this because of consent flows
 ```
 
This information should be available to you already from the web side when you get the access and refresh tokens.